[Fix imjp94/gd-YAFSM#14] Add transitions priority

adamscott opened the pull request:
Closes #14 

The new commit cleans up the code and is partly based on your comments.

The rest of the new codes adds priorities to the existing interface using a spinbox.
![Capture d’écran 2021-10-17 114513](https://user-images.githubusercontent.com/270928/137635030-e7fc87cb-419a-480c-9aae-8b5321f485ff.png)

Thanks for your addon!

imjp94 commented:
Great works! The inspector looks really clean with the spinbox.

Just 2 more modifications before merging:
- There's a bug at `StateMachine.gd` line 52, where `sort_custom()` is called by `from_transitions` which is a dictionary. A possible fix might look like this:
```gdscript
var sorted_from_transitions = from_transitions.values()
sorted_from_transitions.sort_custom(Transition, "sort")
		
for transition in sorted_from_transitions:
    ...
```
- Clean the commits. There's 2 ways I can think of, I will leave it up to you:
  - Just rebase and squash 2 commits into 1
  Or
  - One commit for "Add transition priority"(changes made to `StateMachine.gd` & `Transition.gd`) & another one for "Add inspector for transition priority"(changes made to all the editor script)

adamscott commented:
Sorry for the issue I let through, I already fixed it when I pushed my update, but I forgot to add the file from my test project.

And I squashed the commits together! 😄

imjp94 commented:
Thanks for the contribution @adamscott!
